query-engine/driver-adapters/executor/src/driver-adapters-manager/d1.ts

```diff
@@ -0,0 +1,94 @@
+import path from 'node:path'
+import * as S from '@effect/schema/Schema'
+import { PrismaD1 } from '@prisma/adapter-d1'
+import { DriverAdapter } from '@prisma/driver-adapter-utils'
+import { getPlatformProxy } from 'wrangler'
+import type { D1Database, D1Result } from '@cloudflare/workers-types'
+
+import { __dirname, runBatch } from '../utils'
+import type { ConnectParams, DriverAdaptersManager } from './index'
+import type { DriverAdapterTag, EnvForAdapter } from '../types'
+import { D1Tables } from '../types/d1'
+
+const TAG = 'd1' as const satisfies DriverAdapterTag
+type TAG = typeof TAG
+
+export class D1Manager implements DriverAdaptersManager {
+  #driver: D1Database
+  #dispose: () => Promise<void>
+  #adapter?: DriverAdapter
+
+  private constructor(private env: EnvForAdapter<TAG>, driver: D1Database, dispose: () => Promise<void>) {
+    this.#driver = driver
+    this.#dispose = dispose
+  }
+
+  static async setup(env: EnvForAdapter<TAG>, migrationScript?: string) {
+    const { env: cfBindings, dispose } = await getPlatformProxy<{ D1_DATABASE: D1Database }>({
+      configPath: path.join(__dirname, "../wrangler.toml"),
+    })
+
+    const { D1_DATABASE } = cfBindings
+
+    /* prisma migrate reset */
+    console.warn('[D1] Resetting database')
+    await migrateReset(D1_DATABASE)
+
+    /* prisma migrate diff */
+    if (migrationScript) {
+      console.warn('[D1] Running database migration script')
+      await migrateDiff(D1_DATABASE, migrationScript)
+    }
+
+    return new D1Manager(env, D1_DATABASE, dispose)
+  }
+
+  async connect({}: ConnectParams) {
+    this.#adapter = new PrismaD1(this.#driver)
+    return this.#adapter
+  }
+
+  async teardown() {
+    await this.#dispose()
+  }
+}
+
+async function migrateDiff(D1_DATABASE: D1Database, migrationScript: string) {
+  // Note: when running a script with multiple statements, D1 fails with
+  // `D1_ERROR: A prepared SQL statement must contain only one statement.`
+  // We thus need to run each statement separately, splitting the script by `;`.
+  const sqlStatements = migrationScript.split(';')
+  const preparedStatements = sqlStatements.map((sqlStatement) => D1_DATABASE.prepare(sqlStatement))
+  await runBatch(D1_DATABASE, preparedStatements)
+}
```

Comment:
I'm thinking that for both test suite, we could simplify this later and use `await db.exec(allStatementsAsText)`
https://developers.cloudflare.com/d1/build-with-d1/d1-client-api/#await-dbexec

But no need to do this now.

Reply:
It fails on the first line because it's a comment 😓 
```
    D1_EXEC_ERROR: Error in line 1: -- CreateTable: SQL code did not contain a statement.
    Error: D1_EXEC_ERROR: Error in line 1: -- CreateTable: SQL code did not contain a statement.
```